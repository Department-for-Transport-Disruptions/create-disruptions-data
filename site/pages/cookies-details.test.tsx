import { cleanup, render } from "@testing-library/react";
import { afterEach, describe, expect, it } from "vitest";
import CookieDetails from "./cookie-details.page";

describe("pages", () => {
    describe("cookieDetails", () => {
        afterEach(cleanup);

        it("should render correctly", () => {
            const { asFragment } = render(<CookieDetails />);
            expect(asFragment()).toMatchSnapshot();
        });
    });
});
