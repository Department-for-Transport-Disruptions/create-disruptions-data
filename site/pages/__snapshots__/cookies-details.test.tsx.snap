// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pages > cookieDetails > should render correctly 1`] = `
<DocumentFragment>
  <div
    className="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
  >
    <p
      className="govuk-phase-banner__content"
    >
      <strong
        className="govuk-tag govuk-phase-banner__content__tag"
      >
        Beta
      </strong>
      <span
        className="govuk-phase-banner__text"
      >
        This is a new service. Help us improve it and
         
        <a
          className="govuk-link"
          href="/feedback"
          id="feedback-link"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          give your feedback (opens in new tab)
        </a>
        .
      </span>
    </p>
  </div>,
  <div
    class="govuk-width-container"
  >
    <button
      class="govuk-back-link mb-7"
      type="button"
    >
      Back
    </button>
    <main
      class="govuk-main-wrapper"
      id="main-content"
    >
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-two-thirds"
        >
          <h1
            class="govuk-heading-xl"
          >
            Details about cookies on the Create Transport Disruption Data Service
          </h1>
          <p
            class="govuk-body"
          >
            The Create Transport Disruption Data Service puts small files (known as ‘cookies’) onto your computer to collect information about how you browse the site. Find out more about the cookies we use, what they’re for and when they expire.
          </p>
          <h2
            class="govuk-heading-m"
          >
            Strictly necessary cookies
          </h2>
          <h3
            class="govuk-heading-s"
          >
            Progress through the tool
          </h3>
          <p
            class="govuk-body"
          >
            When you use the Create Transport Disruption Data Service we will set the following cookies as you progress through the forms.
          </p>
          <table
            class="govuk-table cookie-detail-table"
          >
            <thead
              class="govuk-table__head"
            >
              <tr
                class="govuk-table__row"
              >
                <th
                  class="govuk-table__header"
                  scope="col"
                >
                  Name
                </th>
                <th
                  class="govuk-table__header"
                  scope="col"
                >
                  Purpose
                </th>
                <th
                  class="govuk-table__header"
                  scope="col"
                >
                  Expires
                </th>
              </tr>
            </thead>
            <tbody
              class="govuk-table__body"
            >
              <tr
                class="govuk-table__row"
              >
                <td
                  class="govuk-table__cell"
                >
                  _csrf
                </td>
                <td
                  class="govuk-table__cell"
                >
                  Used to secure form submissions
                </td>
                <td
                  class="govuk-table__cell"
                >
                  When you close your browser
                </td>
              </tr>
            </tbody>
          </table>
          <h3
            class="govuk-heading-s"
          >
            Cookies message
          </h3>
          <p
            class="govuk-body"
          >
            You may see a banner when you visit GOV.UK inviting you to accept cookies or review your settings. We’ll set cookies so that your computer knows you’ve seen it and not to show it again, and also to store your settings.
          </p>
          <table
            class="govuk-table cookie-detail-table"
          >
            <thead
              class="govuk-table__head"
            >
              <tr
                class="govuk-table__row"
              >
                <th
                  class="govuk-table__header"
                  scope="col"
                >
                  Name
                </th>
                <th
                  class="govuk-table__header"
                  scope="col"
                >
                  Purpose
                </th>
                <th
                  class="govuk-table__header"
                  scope="col"
                >
                  Expires
                </th>
              </tr>
            </thead>
            <tbody
              class="govuk-table__body"
            >
              <tr
                class="govuk-table__row"
              >
                <td
                  class="govuk-table__cell"
                >
                  cdd-cookies-policy
                </td>
                <td
                  class="govuk-table__cell"
                >
                  Saves your cookie consent settings
                </td>
                <td
                  class="govuk-table__cell"
                >
                  1 year
                </td>
              </tr>
              <tr
                class="govuk-table__row"
              >
                <td
                  class="govuk-table__cell"
                >
                  cdd-cookie-preferences-set
                </td>
                <td
                  class="govuk-table__cell"
                >
                  Lets us know that you’ve saved your cookie consent settings
                </td>
                <td
                  class="govuk-table__cell"
                >
                  1 year
                </td>
              </tr>
            </tbody>
          </table>
          <h2
            class="govuk-heading-m"
          >
            Change your settings
          </h2>
          <p
            class="govuk-body"
          >
            You can 
            <a
              class="govuk-link"
              href="/cookies"
            >
              change which cookies you’re happy for us to use
            </a>
            .
          </p>
        </div>
      </div>
    </main>
    <div>
      <h2
        class="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        class="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link: 
        <a
          class="govuk-link govuk-!-font-size-19"
          href="/contact"
        >
          Contact us
        </a>
      </p>
    </div>
  </div>
  <footer
    class="govuk-footer"
    role="contentinfo"
  >
    <div
      class="govuk-width-container"
    >
      <div
        class="govuk-footer__meta"
      >
        <div
          class="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            class="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            class="govuk-footer__inline-list"
          >
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Help
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/cookies"
              >
                Cookies
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/contact"
              >
                Contact
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/accessibility"
              >
                Accessibility
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
              >
                Terms and conditions
              </a>
            </li>
            <li
              class="govuk-footer__inline-list-item"
            >
              <a
                class="govuk-footer__link"
                href="/privacy"
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            class="govuk-footer__body"
          >
            Built by the 
            <a
              class="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            class="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            class="govuk-footer__licence-description"
          >
            All content is available under the 
            <a
              class="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          class="govuk-footer__meta-item"
        >
          <a
            class="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
          >
            © Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
